static/js/fieldUpdater.js

```diff
@@ -1,53 +1,81 @@
 export default function initialise(options) {
     // gets a reference to the containing element for the component
-    const container = document.querySelector('#' + options.prefix + '-' + options.instance_id);
-
-    const display = container.querySelector('.' + options.prefix + '-display');
-    const inputs = container.querySelector('.' + options.prefix + '-inputs');
-    const input = container.querySelector('.' + options.prefix + '-input');
-    const clear = container.querySelector('.' + options.prefix + '-clear');
-    const submit = container.querySelector('.' + options.prefix + '-submit');
-    const cancel = container.querySelector('.' + options.prefix + '-cancel');
-    const loader = container.querySelector('.' + options.prefix + '-loader');
-    const error = container.querySelector('.' + options.prefix + '-error');
-
-
-    display.onclick = function(e) {
-        inputs.style.display = 'block';
-        display.style.display = 'none';
-        input.value = display.innerHTML;
-        error.style.display = 'none';
-    };
-    clear.onclick = function(e) {
-        alert('clearing');
+    const containerElement = document.querySelector('#' + options.prefix + '-' + options.instance_id);
+
+    // gets references to internal elements
+    const displayElement = containerElement.querySelector('.' + options.prefix + '-display');
```

Comment:
Consider switching all these over to use `getElementById` for better performance.

Reply:
The reason I chose to get by class was dues to browsers annoying habit of registering all elements with ids as variables in the global javascript scope. Which might well have a wider affect on performance, as well as being unpleasant